vision/objectdetection/detection_source.go

```diff
@@ -96,10 +100,19 @@ func (s *Source) Read(ctx context.Context) (image.Image, func(), error) {
 	ctx, span := trace.StartSpan(ctx, "vision::objectdetection::Source::Read")
 	defer span.End()
 	start := time.Now()
+
 	res, err := s.NextResult(ctx)
 	if err != nil {
 		return nil, nil, err
 	}
+
+	for res == nil {
+		res, err = s.NextResult(ctx)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+
```

Comment:
[q] Do you still need this for loop? Is it ever going to produce a nil result?

Reply:
Naw I guess we don't... removing now. Gonna run the test a few thousand times again but you're right.